Makefile

```diff
@@ -139,7 +139,9 @@ ifeq ("$(TRAVIS_COVERAGE)", "1")
 else
 	@echo "Running in native mode."
 	@export log_level=info; export TZ='Asia/Shanghai'; \
-	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -cover $(PACKAGES) -check.p true -check.timeout 4s || { $(FAILPOINT_DISABLE); exit 1; }
+	$(GOTEST) -ldflags '$(TEST_LDFLAGS)' $(EXTRA_TEST_ARGS) -v -cover $(PACKAGES) -check.p true > gotest.log || { $(FAILPOINT_DISABLE); exit 1; }
```

Comment:
I'd suggest we have a dedicate pass to implement this. Otherwise what if tests failed and we cannot find the output from stdout directly?

May we build a check-timeout target base on gotest and add a variable control `-v`

cc @zhouqiang-cl

Reply:
I have tried to make such rule:

```
gotest.log: gotest

gotest:
    go test -v $(PACKAGES) > gotest.log

check-timeout: gotest.log
    grep '^PASS:' gotest.log | go run script.go
```

The problem is, the check-timeout rule will re-run the gotest rule every time, even the gotest.log file exist after running gotest ... @tisonkun 

And when the test fails, how the output is collect before? @zhouqiang-cl

Reply:
Well. Let me think of it...

Another problem is that `-v` for every test may create a large number of logs output, it seems when test failed, digging output is harder, while what we want is just the time.

If our test run less than 10 minutes, I think it is ok we run tests twice so that we don't worry about these things. We get a normal gotest and later go test -v and filter the output. Unrelated output is just discarded.

`go test -v ./... || { failpoint ... } | grep '^PASS:' ...` I mean.

Reply:
But this will lost the original log ...
Some time the test is not stable, and can not reproduce if we run again.

Reply:
You can keep the log if a big log file isn't a concern. Unstable doesn't seem a reason - we'd either stabilize it or skip edge cases.

I'm ok with current solution as a quick prototype only if it doesn't get fail test log lost.

Reply:
I change it a bit, now we can display the log on fail